### PR TITLE
Internal: Docs for JS API - Redux [ED-5711]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

**Review by Nevoss — CHANGES_REQUESTED** [review]
Add React examples + plain JS example

**Review by Nevoss — APPROVED** [review]

File: docs/core/common/assets/js/api/core/store.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   It's actually a wrapper around [Redux Toolkit](https://redux-toolkit.js.org/introduction/getting-started), and it uses [Slices](https://redux-toolkit.js.org/api/createslice) under the hood.
+*  **Location**: *core/common/assets/js/api/core/store.js*
+*  **Methods**:
+
+| Method                        | Params                                                    | Returns                 | Description
+|-------------------------------|-----------------------------------------------------------|-------------------------|---------------------------------------------------------------------
+| `$e.store.register()`         | `{String}` *sliceId*, `{Slice}` *instance*   			    | `{void}`                | Register a new [Redux Toolkit Slice](https://redux-toolkit.js.org/api/createslice).
+| `$e.store.getAll()`           |                                                           | `{Object}`              | Get all existing Slices.
+| `$e.store.get()`              | `{String}` *sliceId* (optional)						    | `{Slice}` \| `{Object}` | Get a specific Slice instance by its ID, or get all of them if ID is empty.
```
Comment (matipojo) [inline docs/core/common/assets/js/api/core/store.md:11]: Why do we need `get all of them if ID is empty`?

File: docs/core/common/assets/js/api/core/store.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+
+## Guidelines & Conventions:
+* Each Slice is owned by a [component](./components.md#guidelines-conventions--files-structure).
+* Each [component](./components.md#guidelines-conventions--files-structure) can override the `defaultStates()` method to return a set of [Slices configurations](https://redux-toolkit.js.org/api/createslice) (e.g. `initialState` & `reducers`).
+* The slices are identified by their key in the Slices definition object, and are prefixed with the component namespace to avoid conflicts (see example below).
+
+
+## Example:
+Defining and consuming a global state inside a component is fairly easy after you understand the basic concepts.
```
Comment (matipojo) [inline docs/core/common/assets/js/api/core/store.md:25]: No `you/we` here and below

**Review by matipojo — CHANGES_REQUESTED** [review]

File: docs/core/common/assets/js/api/core/store.md (hunk 1 above)
Comment (StyleShit) [reply docs/core/common/assets/js/api/core/store.md:11]: We need a way to get all slices. `getAll()` returns an array of strings (like the other JS APIs).
Do you want to add a `getAllSlices()`?

File: docs/core/common/assets/js/api/core/store.md (hunk 2 above)
Comment (StyleShit) [reply docs/core/common/assets/js/api/core/store.md:25]: Every docs out there uses `you / we` when writing...

1. https://reactjs.org/docs/getting-started.html

> If **you’re** interested in playing around with React, **you** can use an online code playground. Try a Hello World template on CodePen, CodeSandbox, or Stackblitz.

> If **you** prefer to use **your** own text editor, **you** can also download this HTML file, edit it, and open it from the local filesystem in **your** browser. It does a slow runtime code transformation, so **we’d** only recommend using this for simple demos



2. https://laravel.com/docs/8.x

> Laravel is a web application framework with expressive, elegant syntax. A web framework provides a structure and starting point for creating **your** application, allowing **you** to focus on creating something amazing while **we** sweat the details.

> Whether **you** are new to PHP or web frameworks or have years of experience, Laravel is a framework that can grow with **you**. **We'll** help you take **your** first steps as a web developer or give **you** a boost as **you** take **your** expertise to the next level. **We** can't wait to see what **you** build.


Anyway, I chaned that...

File: docs/core/common/assets/js/api/core/store.md (hunk 1 above)
Comment (matipojo) [reply docs/core/common/assets/js/api/core/store.md:11]: Yes, please add a `getAllSlices` method.

Comment (StyleShit) [reply docs/core/common/assets/js/api/core/store.md:11]: See #17787

**Pull request merged**